node/actors/bft/src/testonly/ut_harness.rs

```diff
@@ -67,6 +68,57 @@ impl UTHarness {
         }
     }
 
+    pub(crate) async fn iterate_next(&mut self) {
+        let leader_commit = self.new_procedural_leader_commit_many().await;
+        self.dispatch_leader_commit(leader_commit).await.unwrap();
+        self.recv_signed()
+            .await
+            .unwrap()
+            .cast::<ReplicaPrepare>()
+            .unwrap();
+    }
+
+    /// Validate protocol liveness in the aftermath of a timeout.
+    ///
+    /// Params refer to the expected values in the next produced `ReplicaPrepare`
+    /// messages for the new iteration.
+    ///
+    /// * `view` - the expected view of the next `ReplicaPrepare`.
+    /// * `high_vote_view` - the expected view of the high vote of the next `ReplicaPrepare`.
+    /// * `high_qc_view` - the expected view of the high qc of the next `ReplicaPrepare`.
+    ///
+    pub(crate) async fn check_recovery_after_timeout(
+        &mut self,
+        view: ViewNumber,
+        high_vote_view: ViewNumber,
+        high_qc_view: ViewNumber,
+    ) {
+        let replica_prepare = self
+            .recv_signed()
+            .await
+            .unwrap()
+            .cast::<ReplicaPrepare>()
+            .unwrap()
+            .msg;
+
+        assert_matches!(
+            replica_prepare,
+            ReplicaPrepare {
+                view: _view,
+                high_vote,
+                high_qc,
+                ..
+            } => {
+                assert_eq!(_view, view);
+                assert_eq!(high_vote.view, high_vote_view);
+                assert_eq!(high_qc.message.view, high_qc_view);
+            }
+        );
```

Comment:
nit: this assert_matches is unnecesary: you can just write the assert_eq! statements directly.